protocol/directory.go

```diff
@@ -26,19 +26,27 @@ func (d *ConiksDirectory) Update(policies merkletree.Policies) {
 	d.pad.Update(policies)
 }
 
-func (d *ConiksDirectory) Register(uname string, key []byte) (*merkletree.TemporaryBinding, int) {
+func (d *ConiksDirectory) LatestEpoch() uint64 {
+	return d.pad.LatestEpoch()
+}
+
+func (d *ConiksDirectory) Register(uname string, key []byte) (
+	*merkletree.AuthenticationPath, *merkletree.TemporaryBinding, int) {
 	// check whether the name already exists
 	// in the directory before we register
-	ap := d.pad.Lookup(uname)
-	if ap.Leaf().Value() != nil {
-		return nil, ErrorNameExisted
+	ap, err := d.pad.Lookup(uname)
+	if err != nil {
+		return nil, nil, ErrorInternalServer
+	}
+	if !ap.Leaf().IsEmpty() {
+		return ap, nil, ErrorNameExisted
```

Comment:
Couldn't a non-empty AP leaf also mean there's a different index that shares the prefix with ``uname``? This would be a different case than ``ErrorNameExisted``, no?